### PR TITLE
added view all tags to the website

### LINKED ISSUE #35: View All Tags
As an admin I would like to see all the available Tags so that I can choose to edit or delete one, or see that I should add a new one.

**Given** an admin is in the app  
**When** they select the `Tag Management` link in the menu  
**Then** they should be directed to a page that lists all the Tag names ordered alphabetically  

> **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

### PR DESCRIPTION
**Pull request by NathanCenter**
#### Changes Made
1. Added TagControler.cs, ` to `Controllers ` directory.
2. Added Tag.cs, ` to `Models` directory.
3.  Added ITagRepository.cs, TagRepository.cs ` to `Repository` directory.
4. added TagMangement.js and TagView.js to Modules directory

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout nc-displayAllTags
    npm start
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
3. Test app functionality.
   When you first log into the website and run your API you should see Tag Manager in the nav bar. When you click on that it should send you to a new page that shows a list of all the tags that are listed in the datebase.
expected outcome is.
    When user clicks on tag manager, then a new page should show with a list of all the tags.
6. View the code file.
   closes #35

### CODE REVIEW

**Review by matt-rich-anderson — APPROVED** [review]
I review the functionality and everything seems to work as expected.

File: Tabloid/client/src/modules/TagView.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { getAllTags } from './TagManagment';
```
Comment (heymonicakay) [inline Tabloid/client/src/modules/TagView.js]: This is a component module, so it should not be in the modules directory. Please move this to the components directory.

File: Tabloid/client/src/components/Header.js [hunk 2]
```diff
@@ -28,6 +29,9 @@ export default function Header({ isLoggedIn }) {
                 <NavLink tag={RRNavLink} to="/">Home</NavLink>
               </NavItem>
             }
+               <NavItem>
+                <NavLink tag={RRNavLink} to="/Tags" >Tag Managment</NavLink>
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Header.js:37]: Make sure to keep your casing consistent... you've got pascal casing here, but in your app views module you've got lowercase

**Review by heymonicakay — CHANGES_REQUESTED** [review]
Please see my comments. Also, please reference the issue tickets related to this PR in your PR description.

File: Tabloid/client/src/modules/TagView.js (hunk 1 above)
Comment (NathanCenter) [reply Tabloid/client/src/modules/TagView.js:1]: Do I need to move TagManagement to modules or TagView ?

**Comment by NathanCenter** [comment]
Fixed changes that Monica made  showed me

1. This is a component module, so it should not be in the modules directory. Please move this to the components directory.
Added that TagManagement.js is now in the modules directory.
3. Make sure to keep your casing consistent... you've got pascal casing here, but in your app views module you've got lowercase
 changed the path from lower case to uppercase on ApplicationViews.js

File: Tabloid/Controllers/TagController.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Tabloid.Repositories;
+
+namespace Tabloid.Controllers
```
Comment (heymonicakay) [inline Tabloid/Controllers/TagController.cs:6]: Should this controller be auth protected? do we need any other auth checks on the route?

**Review by matt-rich-anderson — APPROVED** [review]
Ito looks like you have made Monica's requested changes

File: Tabloid/Repositories/TagRepository.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Configuration;
+using System.Collections.Generic;
+using Tabloid.Models;
+
+namespace Tabloid.Repositories
```
Comment (heymonicakay) [inline Tabloid/Repositories/TagRepository.cs:7]: Are there DbUtils methods that could be used in this repo to do some of this work? We'll want to use those utils methods consistently across the codebase.

**Review by heymonicakay — CHANGES_REQUESTED** [review]

File: Tabloid/client/src/components/TagManagment.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const baseUrl = 'api/Tag';
```
Comment (heymonicakay) [inline Tabloid/client/src/components/TagManagment.js]: This file does not a component. This file should be in the modules directory.

File: Tabloid/client/src/modules/TagView.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { getAllTags } from '../components/TagManagment';
```
Comment (heymonicakay) [inline Tabloid/client/src/modules/TagView.js]: This file contains a component. This file should be in the components directory.

File: Tabloid/client/src/components/TagManagment.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+export const getAllTags = () => {
+  return fetch(baseUrl)
+    .then((res) => res.json())
+};
+
+export const addTag = (tag) => {
```
Comment (heymonicakay) [inline Tabloid/client/src/components/TagManagment.js]: is addTag supported by your API yet? if not, remove this function from this PR.

File: Tabloid/client/src/components/TagManagment.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+const baseUrl = 'api/Tag';
+
+export const getAllTags = () => {
```
Comment (heymonicakay) [inline Tabloid/client/src/components/TagManagment.js]: if this route is auth protected, you'll need to get an api token to pass in to this fetch call.

**Review by heymonicakay — CHANGES_REQUESTED** [review]
Please see my comments

File: Tabloid/client/src/components/TagManagment.js (hunk 5 above)
Comment (NathanCenter) [reply Tabloid/client/src/components/TagManagment.js]: @heymonicakay Hey I changed the tag management and move it to modules

File: Tabloid/client/src/modules/TagView.js (hunk 6 above)
Comment (NathanCenter) [reply Tabloid/client/src/modules/TagView.js]: @heymonicakay  Hey I changed the tagView and moved it to the components directory

File: Tabloid/client/src/components/TagManagment.js (hunk 8 above)
Comment (NathanCenter) [reply Tabloid/client/src/components/TagManagment.js]: @heymonicakay  Added api token class to the tag mangement!

File: Tabloid/client/src/components/TagManagment.js (hunk 7 above)
Comment (NathanCenter) [reply Tabloid/client/src/components/TagManagment.js]: @heymonicakay got rid of add tag on the tag mangment.js

**Review by heymonicakay — APPROVED** [review]

**Pull request merged**